File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        //Step 1 : Sort the Array
        Arrays.sort(nums);
        //Step 2: Put into ArrayList
        List<Integer> result = new ArrayList<>();
        //Step 3: Iterate through the entire Array
        for(int i = 1;i<nums.length; i++){
            //Step 5: If current element equals previous element, it's a duplicate
            if(nums[i]==nums[i-1]){
                result.add(nums[i]);// Add every duplicate element 
            }
        }
        return result;// return the duplicate elememt from the array
    }
}

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.py
class Solution(object):
    def findDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        #sort the array
        nums.sort()
        #create result list
        result = []
        for i in range(1,len(nums)):
            if nums[i] == nums[i-1]:
                result.append(nums[i])
        return result
        